docs/database/database-spans.md

```diff
@@ -83,9 +83,16 @@ For example, for an operation describing SQL query on an anonymous table like `S
 
 This span describes database client call.
 
-It SHOULD cover the duration of the corresponding call as if it was observed by the caller (such as client application).
-For example, if a transient issue happened and was retried within this database call, the corresponding span should cover the duration of the logical operation
-with all retries.
+Instrumentations SHOULD, when possible, record database spans that cover the duration of
+the corresponding API call as if it was observed by the caller (such as client application).
+For example, if a transient issue happened and was retried within this database call, the corresponding
+span should cover the duration of the logical operation with all retries.
+
+When a database client provides higher-level convenience APIs for specific operations
+(e.g., calling a stored procedure), which internally generate and execute a generic query,
+it is RECOMMENDED to instrument the higher-level convenience APIs.
+These often allow setting `db.operation.*` attributes, which usually are not
+readily available at the generic query level.
 
 **Span name** is covered in the [Name](/docs/database/database-spans.md#name) section.
```

Comment:
we might need to update the span name (to account for missing `db.query.summary` on stored procedures), discussion in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/2693#discussion_r2067819756

Reply:
will follow up on this in a separate PR